src/components/ActionButton/ActionButton.tsx

```diff
@@ -95,23 +94,29 @@ const ActionButton = ({
 
     if (!loading && showLoader) {
       const now = new Date();
-      // calculate elasped time of loading from user and do subtraction to LOADER_MIN_DURATION,
-      // also add an edge case when time diff is less than 0 to be 0.
+      // Keep track of time when loading is occured.
       const loadingMilliseconds: number =
         now.getTime() - (startLoadTime.current ?? now).getTime();
-      loadingTimeTracker.current = Math.max(
+
+      // also add an edge case when time diff is less than 0 to be 0.
+      const loadingTimeout = Math.max(
```

Comment:
Looks like something got jumbled a bit with the comments here:
```suggestion
      // calculate elapsed loading time
      const loadingMilliseconds: number =
        now.getTime() - (startLoadTime.current ?? now).getTime();

      // and subtract it from LOADER_MIN_DURATION,
      // also add an edge case when time diff is less than 0 to be 0.
      const timeoutDuration = Math.max(
```